### PR TITLE
Add some new/updated iuc tools

### PR DESCRIPTION
**Pull request by abretaud**
A little PR to add the tools that got merged in IUC recently
I'd like to update to 17.09 too, I've a PR there on the base image: https://github.com/bgruening/docker-galaxy-sequence-tools/pull/1

As this image is becoming quite huge, and as I use most of the time only the gmod tools, I was thinking about making a `gmod` branch with only apollo, jbrowse, tripal, tools, what do you think?

### CODE REVIEW

**Comment by bgruening** [comment]
Either a branch or a complete new flavor. As you like it makes a lot of sense.
In the long run, we will distribute the dependencies via CVMFS to keep the flavors small.

**Comment by abretaud** [comment]
Yep CVMFS looks great! I think I'll make a branch for now and see if it's still needed once CVMFS is ready

File: genome_annotation_tools.yml [hunk 1]
```diff
@@ -101,9 +101,9 @@ tools:
   owner: bgruening
   tool_panel_section_label: "Gene Prediction"
 
-- name: glimmer3
-  owner: bgruening
-  tool_panel_section_label: "Gene Prediction"
+#- name: glimmer3
```
Comment (bgruening) [inline genome_annotation_tools.yml]: This one should have conda dependencies now and are hosted under IUC. (the owner is still bgruening)

File: genome_annotation_tools.yml [hunk 2]
```diff
@@ -73,9 +73,9 @@ tools:
   owner: peterjc
   tool_panel_section_label: "Classification"
 
-- name: weblogo3
-  owner: devteam
-  tool_panel_section_label: "Visualisation"
+#- name: weblogo3
```
Comment (bgruening) [inline genome_annotation_tools.yml]: we should get this into conda

**Comment by abretaud** [comment]
Do you know if it's possible to get more log from install-tools? It's hard to find out exactly what makes the install fail for each tool

**Comment by abretaud** [comment]
ok, now weblogo3 and glimmer are ok, *but* the job is timing out at 50 minutes which seems to be the travis job [time limit](https://docs.travis-ci.com/user/customizing-the-build#Build-Timeouts) :/
Do you have any idea?

**Comment by bgruening** [comment]
Can you simply write support@travis-ci.com and ask kindly for an extended build time. Usually this works.

File: genome_annotation_tools_2.yml [hunk 3]
```diff
@@ -65,4 +65,19 @@ tools:
 - name: suite_askomics
   owner: gga
   tool_panel_section_label: "Askomics"
-  
+
```
Comment (bgruening) [inline genome_annotation_tools_2.yml]: circos is also available from the TS nowadays, isn't it=

File: install_tools_wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
```
Comment (bgruening) [inline install_tools_wrapper.sh]: Why is this file needed again?

Comment (abretaud) [reply install_tools_wrapper.sh]: I plan to remove it when this PR becomes green
The -v option was also added in https://github.com/bgruening/docker-galaxy-stable/pull/403/files, but I was just too lazy to patch the 17.09 image too + update the bgruening/galaxy-sequence-tools:17.09 image

**Comment by abretaud** [comment]
I'm emailed Travis, waiting for their answer now :)

**Comment by abretaud** [comment]
Ok, now we have a 90minutes timeout... but it's still not enough :/
Maker is taking ages (>30min) to install, I don't really understand why, do you have any other idea apart from asking for more time?

**Comment by bgruening** [comment]
Mh, for me it takes: `conda create maker -n maker2  205,00s user 18,56s system 48% cpu 7:43,73 total`

**Comment by bgruening** [comment]
500MB, that is great to know!
Not sure what this is :(
```
Installing tool suite_snap from iuc to section "Gene Prediction" at revision c24c02e4fcff (TRT: 0:03:21.359972)
	* Error installing a tool (after 0:02:14.575133 seconds)! Name: suite_snap,owner: iuc, revision: c24c02e4fcff, error: {"err_msg": "Uncaught exception in exposed API method:", "err_code": 0}
(3/3) Installing tool suite_maker from iuc to section "Gene Prediction" at revision 90a99e03e5fe (TRT: 0:05:36.104812)
	* Error installing a tool (after 0:00:01.346791 seconds)! Name: suite_maker,owner: iuc, revision: 90a99e03e5fe, error: {"err_msg": "Uncaught exception in exposed API method:", "err_code": 0}
```

**Comment by abretaud** [comment]
I guess it's still failing because the disk is full
I'll see if I can save a little more space, I'm taking some time to experiment ;) (but the image will be huge anyway!)

**Comment by bgruening** [comment]
I have the same problem with the RNA workbench. It gets all to large. Is quay.io building it?

**Comment by abretaud** [comment]
I don't know yet for quay.io, I'm trying to make the tests pass on travis first (but maybe I'll just give up in the end!)

**Comment by abretaud** [comment]
An interesting thing i found today (maybe you already knew it!): when conda creates an env, it creates hardlinks to save disk space. But it only works for files in the same Docker layer. Otherwise, it becomes a file copy => uses more disk space.
That's why I moved augustus, snap and maker to the same tool list => it avoids duplicating the 3.4Gb maker package
Last build went a little more greenish (but still not completely green!)

**Comment by bgruening** [comment]
Oh I didn't know this, interesting so we actually need to think how we structure the layers.
Thanks for sharing!

**Comment by abretaud** [comment]
OMG it's green!
OMG

I ended up modifying .travis.yml to remove all conda envs once a tool list is installed successfully. It's ugly, but I think it's the only way to keep enough disk space in the VM to finish building the image and the run the tests.

I have to leave now, if you're ok to keep this little hack, I'll clean the mess I made in this PR tomorrow before merging

**Comment by bgruening** [comment]
@abretaud can you document the purpose of the hack in the travis yaml, so we remember why this was needed. Its great got this working, I'm not a fan of this but I also don't see any other way currently. We need to have the tools CVMFS at some point.

Thanks a lot!

**Comment by abretaud** [comment]
Oh damn, there is even less disk space on quay.io... I contacted them just in case

**Comment by bgruening** [comment]
@abretaud quay.io can store >20GB, isn't it?

**Comment by abretaud** [comment]
It seems like they can store images >20Gb, but not build them as the builder has ~30Gb total disk space
As it's not super urgent for me I'll wait a little to see if they find a solution (ticket escalated at the moment)

**Comment by abretaud** [comment]
I got a final answer from quay.io: as I expected there's no way to add more disk space to the builders.

So, until we have CVMFS, I think we have the choice between:
1. adding a "install_resolver_dependencies: False" to the tools that take a lot of space (like maker, busco...). As `GALAXY_CONFIG_CONDA_AUTO_INSTALL` is `True`, the deps would still be installed when people launch these tools
2. pushing manually the image to quay.io

Do you have a preference?
I'm tempted to go for 1 as people will probably not like downloading a 22Gb image

**Comment by abretaud** [comment]
(I guess you've been notified: the much smaller `quay.io/galaxy/genome-annotation:gmod` image is online)

**Comment by bgruening** [comment]
1)!

And here is some hope: https://depot.galaxyproject.org/singularity/
